src/Compilers/Core/Portable/Compilation/CompilationOptions.cs

```diff
@@ -233,6 +233,15 @@ public abstract class CompilationOptions
         /// </summary>
         public AssemblyIdentityComparer AssemblyIdentityComparer { get; protected set; }
 
+        /// <summary>
+        /// Gets the default nullable context state in this compilation.
+        /// </summary>
+        /// <remarks>
+        /// This context does not apply to files that are marked as generated. Nullable is off
+        /// by default in those locations.
+        /// </remarks>
+        public abstract NullableContextOptions NullableContextOptions { get; protected set; }
```

Comment:
I'd make the setter `private protected`, but this is really a dumb restriction that we can't add a setter in an override.